### DISCUSSION TITLE
helm chart

### DISCUSSION THREAD

**Discussion by ehsanjavadynia** [misskey-dev/misskey#13989] [General]
Hi, is there any helm chart for kubernetes setup?

**Comment by ghost** [comment]
@ehsanjavadynia did you progress on a kubernetes setup with misskey?

**Comment by ssmucny** [comment]
I am working on a helm chart for misskey using [True Charts](https://github.com/truecharts/charts). Got the basic helm install working and am looking at TrueNAS app integration next. 

https://github.com/ssmucny/charts/tree/misskey/charts/incubator/misskey

**Comment by mi-24v** [comment]
see https://github.com/misskey-dev/misskey/tree/develop/chart
misskey was set up with [Okteto](https://www.okteto.com/) when [#8684](https://github.com/misskey-dev/misskey/pull/8684) was merged

**Comment by ssmucny** [comment]
Misskey is now on the TrueCharts helm chart repository in the incubator train! It is designed to be used with [TrueNAS Scale](https://www.truenas.com/truenas-scale/) but can be installed as a normal helm chart as well. Because it is in incubator, it is considered to be in "beta" still, and may change as issues are found and resolved. With enough positive feedback, Misskey can be promoted to the stable train which includes proper support from the TrueCharts team.

[Misskey on TrueCharts](https://truecharts.org/docs/charts/incubator/misskey/)
[Incubator Info](https://truecharts.org/docs/manual/SUPPORT/#incubator-train-charts)
[Merged PR](https://github.com/truecharts/charts/pull/4315)
[Merge Commit](https://github.com/truecharts/charts/commit/bcd3cb6c09ece36a725877c831c3e48729827e82)

**Comment by ssmucny** [comment]
Version has been upgraded to 1.0.2, and I have been able to successfully install it. I can add English documentation on the Misskey hub repo for helm install, but otherwise this issue can be closed.

**Comment by saza-ku** [comment]
Truecharts no longer supports native helm charts.  
So we can't install the chart in Kubernetes.

ref: https://twitter.com/useTrueCharts/status/1592552717503041536?s=20&t=Ns-zHtiDK3oiPV9ZG-_fIw

**Comment by ssmucny** [comment]
> Truecharts no longer supports native helm charts. So we can't install the chart in Kubernetes.
> 
> ref: https://twitter.com/useTrueCharts/status/1592552717503041536?s=20&t=Ns-zHtiDK3oiPV9ZG-_fIw

Looks like the new pipeline doesn't build the charts outside of dependency and common, so you can't even use it unofficially. This was a result of changing their build pipeline (reduced the build time from 12 hours to 20 minutes).

The plain Helm install method should be removed and just leave the TrueCharts instillation method. For people that want to roll their own Helm install, currently they will have to manually build their cluster by modifying the chart files using https://github.com/misskey-dev/misskey/tree/develop/chart directory and https://github.com/truecharts/charts/tree/master/charts/incubator/misskey as examples.
